earth_enterprise/src/fusion/asset-tests/bin/generate_test_asset_xmls.py

```diff
@@ -9,13 +9,73 @@
 file operations in the test assets directory slow.
 """
 
+import argparse
+import functools
 import jinja2
+import math
 import os
 import os.path
 import sys
 import uuid
 
 
+# Create some decorators for methods that serve as tasks to be executed.
+def execute_once(func):
+    """Make a function exectule only once.  Calls after the first one return
+    immediately.
+    """
+
+    @functools.wraps(func)
+    def wrapper_execute_once(*args, **kwargs):
+        if wrapper_execute_once.executed:
+            return
```

Comment:
should you save res and return it here?

Reply:
This is only used for functions that don't return a meaningful value, and this return value might tell you the function didn't execute.

There's already a library that implements memoization, which cache the return results for each set of input parameters, but I thought that was more than was needed in this case, so I just I just wrote this short decorator.

If this is confusing, I'll just switch over to the library decorator.